packages/session-replay-browser/src/session-replay.ts

```diff
@@ -81,7 +94,17 @@ export class SessionReplay implements AmplitudeSessionReplay {
     this.stopRecordingAndSendEvents(this.identifiers.sessionId);
     this.identifiers.sessionId = sessionId;
     this.eventsManager && this.eventsManager.resetSequence();
-    this.recordEvents();
+    if (this.joinedConfigGenerator) {
+      this.joinedConfigGenerator
+        .generateJoinedConfig(this.identifiers.sessionId)
+        .then((updatedConfig) => {
```

Comment:
Do we want to eventually refactor this to an async method?

Reply:
https://amplitude.atlassian.net/browse/AMP-98766